modules/spring-config/src/main/java/org/mule/config/spring/SpringRegistryBootstrap.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.config.spring;
+
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.api.MuleException;
+import org.mule.api.lifecycle.InitialisationException;
+import org.mule.api.registry.Registry;
+import org.mule.api.registry.RegistryProvider;
+import org.mule.api.transformer.DataType;
+import org.mule.api.transformer.Transformer;
+import org.mule.config.bootstrap.BootstrapObjectFactory;
+import org.mule.config.bootstrap.ClassPathRegistryBootstrapDiscoverer;
+import org.mule.config.bootstrap.SimpleRegistryBootstrap;
+import org.mule.config.spring.factories.BootstrapObjectFactoryBean;
+import org.mule.config.spring.factories.ConstantFactoryBean;
+import org.mule.transformer.TransformerUtils;
+import org.mule.transformer.types.DataTypeFactory;
+import org.mule.util.ClassUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map.Entry;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.context.ApplicationContext;
+
+/**
+ * Specialization of {@link SimpleRegistryBootstrap which instead of registering the objects directly
+ * into a {@link Registry}, generates {@link BeanDefinition}s into a {@link BeanDefinitionRegistry} so
+ * that the Spring framework can create those objects when initialising an {@link ApplicationContext}}
+ *
+ * @since 3.7.0
+ */
+public class SpringRegistryBootstrap extends SimpleRegistryBootstrap implements BeanFactoryAware
```

Comment:
This class is extending a deprecated class. Would be better to have an abstract class containing the shared behaviour